src/nomad_simulations/properties/band_gap.py

```diff
@@ -0,0 +1,147 @@
+#
+# Copyright The NOMAD Authors.
+#
+# This file is part of NOMAD. See https://nomad-lab.eu for further info.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import numpy as np
+from structlog.stdlib import BoundLogger
+import pint
+from typing import Optional
+
+from nomad.units import ureg
+from nomad.metainfo import Quantity, MEnum, Section, Context
+
+from ..physical_property import PhysicalProperty
+
+
+class ElectronicBandGap(PhysicalProperty):
+    """
+    Energy difference between the highest occupied electronic state and the lowest unoccupied electronic state.
+    """
+
+    iri = 'https://fairmat-nfdi.github.io/fairmat-taxonomy/#/ElectronicBandGap'
+
+    # ? can `type` change character depending on the `variables`?
```

Comment:
Can you give me an example?

Reply:
Ow, sorry, I misread before: I didn't take in `variables` properly, maybe call it `Variable` for clarity.
Not sure whether you mean the actual stored type, or the range of allowed types. In case of the latter, I think we should aim to cover all options regardless of the variables.

Reply:
I was putting myself in the following situation:

Band gap varies with X. At some critical Xc, it changes `type` from `'direct'` to `'indirect'`. However, I don't know if such a variable X might exist.

I was thinking that anyways we should `type` and other id strings as general global quantities. I.e., when these become list they lost all their usefulness.

Reply:
> Band gap varies with X. At some critical Xc, it changes type from 'direct' to 'indirect'. However, I don't know if such a variable X might exist.

Rather than the variable, wouldn't your example more so depend on the conditions under which the band gap was measured / computed?

Anyhow, I agree that `type` as a global quantity is fine.